students/Sally_Shen/lesson7/run_html_render.py

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env python3
+
+"""
+a simple script can run and test your html rendering classes.
+
+Uncomment the steps as you add to your rendering.
+
+"""
+
+from io import StringIO
+
+# importing the html_rendering code with a short name for easy typing.
+import html_render as hr
+
+
+# writing the file out:
+def render_page(page, filename, indent=None):
+    """
+    render the tree of elements
+
+    This uses StringIO to render to memory, then dump to console and
+    write to file -- very handy!
+    """
+
+    f = StringIO()
+    if indent is None:
+        page.render(f)
+    else:
+        page.render(f, indent)
+
+    print(f.getvalue())
+    with open(filename, 'w') as outfile:
+        outfile.write(f.getvalue())
+
+
+# Step 1
```

Comment:
Don't forget to comment this out and run it!

Reply:
Sure, I finished all the test but only commented/uncommented step by step, will uncomment all.